packages/application-shell/src/utils/get-mc-api-url/get-mc-api-url.ts

```diff
@@ -0,0 +1,19 @@
+import { ApplicationWindow } from '@commercetools-frontend/constants';
+
+declare let window: ApplicationWindow;
+
+const getMcApiUrlFromOrigin = () => {
+  const url = new URL(window.origin);
+  const mcApiUrlHost = url.host.replace('mc.', 'mc-api.');
+
+  return `${url.protocol}//${mcApiUrlHost}`;
+};
+
+export default function getMcApiUrl(appEnv = window.app) {
+  const isServedByProxy = JSON.parse(appEnv.servedByProxy);
+  const mcApiUrl = appEnv.mcApiUrl;
```

Comment:
Hmm I don't think we need to make this optional. We should just check that the app is running behind the proxy. For custom apps that's not a problem, they also run on the same domain in production.

Reply:
> We should just check that the app is running behind the proxy

...to then what. The intend here is a bit unclear to me. Can you elaborate? You man everything running behind the proxy always gets an MC API url from the origin?